### ISSUE TITLE
Wrong code snippets printed for errors in `--import-ast` mode

### ISSUE THREAD

**Issue by cameel** [ethereum/solidity#12168]
## Description
If you save the AST using `solc --combined-json ast` and then load it with `solc --import-ast`, the code snippets included in errors printed to the console show bits of the JSON input rather than actual code.

Since the original source is not available in the AST import mode, the only solution seems to be to show only file name and line number.

## Environment

- Compiler version: 0.8.10 (`develop`)

## Steps to Reproduce
```bash
echo 'contract C { function foo(uint x) public { require(x > 0); } }' \
    solc --combined-json ast - | \
    solc --import-ast -
```
Output (first `solc` command):
```
Warning: Function state mutability can be restricted to pure
 --> <stdin>:1:14:
  |
1 | contract C { function foo(uint x) public { require(x > 0); } }
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Output (second `solc` command):
```
Warning: Function state mutability can be restricted to pure
 --> <stdin>:1:14:
  |
1 | {"absolutePath":"<stdin>","exportedSymbols":{"C":[13]},"id":14,"nodeType":"SourceUnit","nodes": ...
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

**Comment by chriseth** [comment]
This is a known limitation, and I think it is better to preserve the debug info than removing it. Also note that json ast import is a very rarely used feature, so we should be careful not to complicate the debug info just because of this.

**Comment by cameel** [comment]
Can't we just print the file name and line number without printing the snippet?

**Comment by chriseth** [comment]
We can, but we first need to know about this situation.

**Comment by cameel** [comment]
It was fixed by #13244.
